src/components/EmailEditor.vue

```diff
@@ -51,8 +51,25 @@ export default defineComponent({
     },
   },
   setup() {
-    // shallowRef is used to avoid window.postMessage error
-    const editor = ref<EmailEditorProps['editor'] | null>(null); // Creates a reactive reference
+    const editor = shallowRef<EmailEditorProps['editor'] | null>(null);
+
+    // Add a wrapper for the editor to handle serialization
+    const createSerializedEditor = (originalEditor: any) => {
+      if (!originalEditor) return null;
+      
+      return {
+        ...originalEditor,
+        registerCallback: (event: string, callback: Function) => {
+          originalEditor.registerCallback(event, (data: any, done: Function) => {
+            callback(data, (result: any) => {
+              // Serialize the result before passing to postMessage
+              const safeResult = JSON.parse(JSON.stringify(result));
```

Comment:
We can fix it on our side, but maybe using `toRaw` instead of `JSON.parse(JSON.stringify())`.